src/runtime/methodbinder.cs

```diff
@@ -652,7 +652,7 @@ static Type TryComputeClrArgumentType(Type parameterType, IntPtr argument, bool
             {
                 match = true;
             }
-            else if (positionalArgumentCount < parameters.Length)
+            else if (positionalArgumentCount < parameters.Length && (!paramsArray || positionalArgumentCount == parameters.Length - 1))
```

Comment:
Maybe it would be useful to comment explicitly on what kind of cases would fall into this condition, or simplify the branching statements somehow (refactor the default arg logic in a separate method)? Just by reading this I'm guessing that this should take care of only the following 2 cases?
1. when it's not a `params` array but there are some default values we need to fill in
2. when it's a `params` array and exactly one parameter is missing 

Maybe there are more?

Reply:
Nevermind, actually I get it now! I also added an extra comment and renamed the variables for readability and consistency with the others parts of methodbinder in my PR: https://github.com/pythonnet/pythonnet/pull/1324/files#diff-237dc718ad5424f5476d2ebbbc4e5164c95901bd2a4c2daaf8ca6fc9f9d3a5fbR703-R704